packages/extension-api/src/extension-api.d.ts

```diff
@@ -650,16 +651,77 @@ declare module '@kortex-app/api' {
     models: Array<InferenceModel>;
   };
 
+  export interface InputResponse {
+    value: string;
+  }
+
+  export interface InputWithVariableResponse extends InputResponse {
+    variables: Record<string, InputResponse>;
+  }
+
+  export interface MCPRemoteServerConfig {
+    type: 'remote';
+    index: number;
+    headers: Record<string, InputWithVariableResponse>;
+  }
+
+  export interface MCPPackageServerConfig {
+    type: 'package';
+    index: number;
+    runtimeArguments: Record<number, InputWithVariableResponse>;
+    packageArguments: Record<number, InputWithVariableResponse>;
+    environmentVariables: Record<string, InputWithVariableResponse>;
+  }
+
+  export type MCPServerConfig = MCPRemoteServerConfig | MCPPackageServerConfig;
+
+  export type MCPServerDetail = components['schemas']['ServerDetail'];
+
+  export type MCPServer = {
+    server: MCPServerDetail;
+    config: MCPServerConfig;
+  };
+
+  export type RagProviderConnection = {
+    name: string;
+    mcpServer: MCPServer;
```

Comment:
question: I don't see why we would have the definition of an MCP server here

issue: I would say it should be the id of the MCP server, not its definition.

AFAIK Registering MCP is done separately